plugin/advisor/pg/advisor_column_type_disallow_list.go

```diff
@@ -0,0 +1,152 @@
+package pg
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/advisor/db"
+	"github.com/bytebase/bytebase/plugin/parser/ast"
+)
+
+func init() {
+	advisor.Register(db.Postgres, advisor.PostgreSQLColumnTypeDisallowList, &ColumnTypeDisallowListAdvisor{})
+}
+
+type ColumnTypeDisallowListAdvisor struct {
+}
+
+func (*ColumnTypeDisallowListAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	stmtList, errAdvice := parseStatement(statement)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySQLReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	payload, err := advisor.UnmarshalStringArrayTypeRulePayload(ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &columnTypeDisallowListChecker{
+		level:           level,
+		title:           string(ctx.Rule.Type),
+		typeRestriction: make(map[string]bool),
+	}
+	for _, tp := range payload.List {
+		checker.typeRestriction[strings.ToLower(tp)] = true
+	}
+
+	for _, stmt := range stmtList {
+		checker.text = stmt.Text()
+		checker.line = stmt.LastLine()
+		ast.Walk(checker, stmt)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+}
+
+type columnTypeDisallowListChecker struct {
+	adviceList      []advisor.Advice
+	level           advisor.Status
+	title           string
+	text            string
+	line            int
+	typeRestriction map[string]bool
+}
+
+type columnTypeData struct {
+	table  string
+	column string
+	tp     string
+	line   int
+}
+
+func (checker *columnTypeDisallowListChecker) Visit(in ast.Node) ast.Visitor {
+	var columnList []columnTypeData
+	switch node := in.(type) {
+	case *ast.CreateTableStmt:
+		for _, column := range node.ColumnList {
+			exist := false
+			typeDisallow := ""
+			for tp, _ := range checker.typeRestriction {
+				if exist = column.Type.EquivalentType(tp); exist {
+					typeDisallow = tp
+					break
+				}
+			}
+			if exist {
+				columnList = append(columnList, columnTypeData{
+					table:  node.Name.Name,
+					column: column.ColumnName,
+					tp:     strings.ToUpper(typeDisallow),
+					line:   checker.line,
```

Comment:
For `columns` in `CREATE TABLE` statements, we set the line for them specifically. So we can use `column.LastLine()` instead `checker.line` here.

Sadly, we only support this for `CREATE TABLE` statements now.